src/Compiler/Utilities/HashMultiMap.fs

```diff
@@ -3,6 +3,17 @@
 namespace Internal.Utilities.Collections
 
 open System.Collections.Generic
+open System.Collections.Concurrent
+
+[<AutoOpen>]
+module HashMultiMap =
+    type IDictionary<'K, 'V> with
+        // We only call it on `.Copy()` so it should be fine.
+        member this.GetComparer() =
+            match this with
+            | :? Dictionary<'K, 'V> as d -> d.Comparer
+            | :? ConcurrentDictionary<'K, 'V> as d -> d.Comparer
```

Comment:
This does not work in ns20, the property was added in net6

Reply:
Maybe the HashMultiMap  will have to store the comparer reference itself.

Reply:
Yeah, I had this in mind as backup plan. Will change it